lib/elsa/producer.ex

```diff
@@ -14,6 +14,15 @@ defmodule Elsa.Producer do
       from the total available topic partitions or assign an integer based on an md5 hash of the messages.
   """
 
+  @typedoc """
+  Elsa messages can take a number of different forms, including a single binary, a key/value tuple, a map
+  including `:key` and `:value` keys, or a list of iolists. Because Elsa supports both single messages and
+  lists of messages and because an iolist is indistinguishable from a list of other message types from the
+  perspective of the compiler, even single-message iolists must be wrapped in an additional list in order to
+  be produced. Internally, all messages are converted to a map before being encoded and produced.
+  """
+  @type message :: {iodata(), iodata()} | binary() | %{key: iodata(), value: iodata()} | [iolist()]
```

Comment:
should this be a lists of iolist()??

I thought it would be just | iolist()

Reply:
if it's not a list of iolists then wouldn't it try to parse that as a list of multiple messages and blow up when they weren't of the form `binary() | map() | tuple()`?